### DISCUSSION TITLE
theory/implementation of stress divergence kernel (Jacobian)

### DISCUSSION THREAD

**Discussion by kteferra2** [idaholab/moose#27900] [Q&A Modules: Solid mechanics]
Hi,
Would anyone be able to provide reference to either an old paper, textbook, or theory manual that best describes the implementation of the finite deformation updated Lagrangian formulation? Specifically, I'm looking to understand the theory and implementation details for the function "StressDivergenceTensors::computeFiniteDeformJacobian()". I would greatly appreciate it if anyone can point me in the right direction. Thanks!
kt

**Comment by GiudGiud** [comment]
Maybe look for technical reports on OSTI produced by the NEAMS program in mechanics / fuels recently

otherwise @hugary1995 will know

**Reply by hugary1995** [reply]
IIRC the old solid mechanics implementation for the "incremental" formulation was following Malvern's textbook "Introduction to the Mechanics of a Continuous Medium". I think some schools still use that textbook, but it's becoming less popular. The `StressDivergenceTensors::computeFiniteDeformJacobian()` predates me, and I suppose @bwspenc would know the context of that function.

For the new system (i.e. with `new_system = true`), there is a journal article under review.
